src/db/collections/outputs/mod.rs

```diff
@@ -213,6 +214,74 @@ impl OutputCollection {
         Ok(())
     }
 
+    /// Create the initial ledger updates materialized view by pulling all output data.
+    #[instrument(skip_all, err, level = "trace")]
+    pub async fn create_ledger_updates(&self) -> Result<(), Error> {
+        self.aggregate::<OutputDocument>(
+            vec![
+                doc! { "$match": { "details.address": { "$ne": null } } },
+                doc! { "$project": {
+                    "_id": {
+                        "milestone_index": { "$ifNull": [
+                            "$metadata.spent_metadata.spent.milestone_index",
+                            "$metadata.booked.milestone_index"
+                        ] },
+                        "output_id": "$_id",
+                        "is_spent": { "$ne": [ "$metadata.spent_metadata", null ] },
+                    },
+                    "address": "$details.address",
+                    "milestone_timestamp": { "$ifNull": [
+                        "$metadata.spent_metadata.spent.milestone_timestamp",
+                        "$metadata.booked.milestone_timestamp"
+                    ] },
+                } },
```

Comment:
As the `$project` stage is identical I would love to pull this out into a function to reduce the redundancy.

Reply:
Let's worry about this as part of our refactoring of these later.

Reply:
I mean pulling out the pipeline stages into functions or whatever.

Reply:
We should grab these low-hanging fruits though, especially if we have already identified areas for improvement.

Reply:
Unfortunately it is not low hanging in my opinion. The right solution is not as clear as simply pulling this into a function. I think we need to give it some thought and apply a uniform solution everywhere.